server/src/BitBakeDocScanner.ts

```diff
@@ -88,4 +96,52 @@ export class BitBakeDocScanner {
     const variablesRegExpString = `(${variablesNames.join('|')})`
     this._variablesRegex = new RegExp(variablesRegExpString, 'gi')
   }
+
+  public parseYoctoTaskFile (): void {
+    let file = ''
+    try {
+      file = fs.readFileSync(yoctoTaskFilePath, 'utf8')
+    } catch {
+      logger.warn(`Failed to read Yocto task file at ${yoctoTaskFilePath}`)
+    }
+
+    const tasks: CompletionItem[] = []
+    for (const yoctoTaskMatch of file.matchAll(yoctoTaskPattern)) {
+      const taskName = yoctoTaskMatch.groups?.name
+      const taskDescription = yoctoTaskMatch.groups?.description
+        .replace(/\\n(?=")/g, '')
+        .replace(/:term:|:ref:/g, '')
+        .replace(/\.\. (note|important)::/g, (_match, p1) => { return `**${p1}**` })
+        .replace(/::/g, ':')
+        .replace(/``/g, '`')
+        .replace(/`\$\{`[\\]+\s(.*)[\\]+\s`\}`/g, (_match, p1) => p1)
+        .replace(/^\n(\s{5,})/gm, '\n\t') // when 5 or more spaces are present as indentation, the texts following are likely code. Replace the indentation with one tab to trigger highlighting in this documentation
+
+      if (taskName !== undefined) {
+        tasks.push({
+          label: taskName,
+          documentation: taskDescription
+        })
+      }
+    }
+
+    // The only one that is not matched, so explicitly add it here
```

Comment:
I suggest not including this in this PR. Instead, we should update the poky docs to be easier to parse by machines!
Why did this task not get parsed properly? It's probably easy to fix in the doc and contribute.

Reply:
As discussed, the fetch approach should come next time, for now, I added a line at the end of `tasks.rst` to let the last item be properly matched.